rules/alerting.go

```diff
@@ -380,7 +380,7 @@ func (r *AlertingRule) Eval(ctx context.Context, queryOffset time.Duration, ts t
 			result, err := tmpl.Expand()
 			if err != nil {
 				result = fmt.Sprintf("<error expanding template: %s>", err)
-				r.logger.Warn("Expanding alert template failed", "err", err, "data", tmplData)
+				r.logger.Warn("Expanding alert template failed", "err", err, "data", fmt.Sprintf("%#v", tmplData))
```

Comment:
Suggest in `template.go` to make `tmplData` a struct defined at package scope and then implement `String()` on that struct.

Reply:
done in https://github.com/prometheus/prometheus/pull/15093/commits/8b45384a44c56158cdc0fb3bd9b7df419b1cf5ac